.drone.yml

```diff
@@ -17,46 +17,52 @@ pipeline:
     # dont clone submodules
     recursive: false
 
-  wait-for-build:
-    image: harbor.ci.drone.local/library/vic-integration-test:1.36
-    pull: true
-    environment:
-      BIN: bin
-      GOPATH: /go
-      SHELL: /bin/bash
-      TEST_URL_ARRAY: ${TEST_URL_ARRAY}
-      DRONE_SERVER: ${DRONE_SERVER}
-      DRONE_TOKEN: ${DRONE_TOKEN}
-    commands:
-      - tests/wait_until_previous_builds_complete.sh
-
   vic-engine:
     image: harbor.ci.drone.local/library/vic-integration-test:1.36
     pull: true
     environment:
       BIN: bin
       GOPATH: /go
       SHELL: /bin/bash
-      TEST_URL_ARRAY:  ${TEST_URL_ARRAY}
+      TEST_URL:  ${TEST_URL}
       TEST_USERNAME:  ${TEST_USERNAME}
       TEST_PASSWORD:  ${TEST_PASSWORD}
       REGISTRY_PASSWORD:  ${REGISTRY_PASSWORD}
       BUILD_NUMBER: ${DRONE_BUILD_NUMBER}
       COMMIT: ${DRONE_COMMIT}
     commands:
+      - . ./tests/ci-env.sh # set CI env variable values
       - make mark
       - make all
       - make sincemark
       - make mark
       - echo `ls vendor/github.com/vmware/govmomi/vim25/methods`
       - echo `ls vendor/github.com/vmware/govmomi/vim25/types`
       - echo `ls vendor/github.com/docker/docker/vendor/github.com/opencontainers/runc/libcontainer/system`
-      - export VIC_ESX_URL_ARRAY="`tests/get_test_url.sh`"
+      - export VIC_ESX_URL_ARRAY="$TEST_USERNAME:$TEST_PASSWORD@$TEST_URL"
       - make test
       - make sincemark
+    volumes:
+      - /home/vic:/ci
     when:
       status:  success
 
+  wait-for-build:
+    image: harbor.ci.drone.local/library/vic-integration-test:1.36
+    pull: true
+    environment:
+      BIN: bin
+      GOPATH: /go
+      SHELL: /bin/bash
+      DRONE_SERVER: ${DRONE_SERVER}
+      DRONE_TOKEN: ${DRONE_TOKEN}
+      ACTIVE_BUILD_NUMBER: ${ACTIVE_BUILD_NUMBER}
```

Comment:
where is this coming from and what is it supposed to represent?

Reply:
`ACTIVE_BUILD_NUMBER` represents the latest build run on that worker and is the additional env var set in the volume mounted file `test_env`, its initial value is `0` but when some job is run on that worker the value is set to that build number so that any other concurrent job running after that can wait until its done.

`test_env` looks like below, which will have 3 variables set.

```
TEST_URL=192.168.31.57
DRONE_MACHINE=vic-drone-05-worker-13
ACTIVE_BUILD_NUMBER=0
```

Reply:
I think this ultimately is going to be actually worse than the existing solution.. right now, every 14th builds will wait until the previous build completes.  With this solution, every other build will wait until it completes, because drone server appears to load up two jobs on a worker at once or in near vicinity of each other as opposed to a true round robin.  In order to get the benefits of this solution, I believe you will need to re-work it into pinning two ESX per worker in test_env and then somehow decide which one is open...

Reply:
Based on my experience so far with Drone, I think it will not run more than one job per worker concurrently unless all the workers are busy with jobs. That means, waiting is triggered only after 15th running job. Also, we haven't set `DRONE_MAX_PROCS` value, so there is no guarantee that only two jobs per worker at once are run, so pinning 2 ESXs won't help.

Reply:
Everytime I have looked in the past couple of weeks, there were 2 job running on 3 or 4 workers and the rest were empty.. If you are sure that the server does in fact do a round robin, then LGTM.

Reply:
Cool, I will verify it before I merge this. Thanks!